src/Akeneo/Connectivity/Connection/back/Infrastructure/Apps/AsymmetricKeysGenerator.php

```diff
@@ -5,38 +5,44 @@
 
 use Akeneo\Connectivity\Connection\Domain\Apps\AsymmetricKeysGeneratorInterface;
 use Akeneo\Connectivity\Connection\Domain\Apps\DTO\AsymmetricKeys;
-use phpseclib\Crypt\RSA;
-use phpseclib\File\X509;
+use phpseclib3\Crypt\RSA;
+use phpseclib3\File\X509;
 
 /**
  * @copyright 2021 Akeneo SAS (http://www.akeneo.com)
  * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
  */
 class AsymmetricKeysGenerator implements AsymmetricKeysGeneratorInterface
 {
-    public function generate(): AsymmetricKeys
+    public function __construct(private string $openSSLConfigPath)
     {
-        $privKey = new RSA();
-        $keys = $privKey->createKey();
-        $privateKey = $keys['privatekey'];
-        $publicKey = $keys['publickey'];
+    }
 
-        $pubKey = new RSA();
-        $pubKey->loadKey($publicKey);
-        $pubKey->setPublicKey();
+    public function generate(): AsymmetricKeys
+    {
+        /*
+         * Following algorithm is the implementation documented by the phpseclib library
+         * in order to generate self-signed public key and private key.
+         * see http://phpseclib.sourceforge.net/x509/guide.html#selfsigned
+         */
+        RSA::setOpenSSLConfigPath($this->openSSLConfigPath);
+        /** @var RSA\PrivateKey $privateKey */
+        $privateKey = RSA::createKey();
+        $publicKey = $privateKey->getPublicKey();
```

Comment:
To be fully compatible with how the public key is used on the app side, we need to add this line of configuration. It seems that in phpseclib3 they moved to a more secure one called PSS. All of this is a bit obscure to me, so don't hesitate to challenge my proposal.
```suggestion
        RSA::setOpenSSLConfigPath($this->openSSLConfigPath);
        /** @var RSA\PrivateKey $privateKey */
        $privateKey = RSA::createKey();
        $privateKey = $privateKey->withPadding(RSA::SIGNATURE_PKCS1);
        $publicKey = $privateKey->getPublicKey();
```

To see a working example, take a look at the demo app that is using `lcobucci/jwt` lib (4.1.5). The error appears in `src/Signer/OpenSSL.php` when trying to do `openssl_pkey_get_details($key)` on the public key. It returns a type `-1` meaning unknown. Adding this line of config makes it return a type `OPENSSL_KEYTYPE_RSA`.